### PR TITLE
Add release actions

### PR DESCRIPTION
**Pull request by Callisto13**
WIP - feel free to add or amend as you like

### Description

This adds some very basic release flow automation.

I have decided to keep things super basic, therefore:
- No release-candidate nonsense
- No branching, only tags
- We don't need to release any binaries, so assets are just crds and pullable docker images (multi-arch)
- Current manual steps added to readme, can be automated further.

These are based on what the flux team do with their component releases. I have set to just push to dockerhub, but there may be other registries we want.

(TODO: set up tokens etc)

### CODE REVIEW

**Comment by aclevername** [comment]
@Callisto13 is this ready for review or still a draft?

**Comment by Callisto13** [comment]
Bit of both. I intend to come back to it after the bootstrapping mess, but you can review/alter/give thoughts etc as you want to if you have time now.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      - 'v*'
+  # should be able to not have to create tag manually
+  # workflow_dispatch: {}
+
+env:
+  CONTROLLER: profiles-controller
+  ORG: weaveworks
+
+jobs:
+  # run all tests first?
```
Comment (aclevername) [inline .github/workflows/release.yaml:14]: if its on main it should be green since it passed PR review :thinking:

Comment (Callisto13) [reply .github/workflows/release.yaml:14]: indeed, one would hope

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            org.opencontainers.image.description=${{ github.event.repository.description }}
+            org.opencontainers.image.url=${{ github.event.repository.html_url }}
+            org.opencontainers.image.revision=${{ github.sha }}
+            org.opencontainers.image.version=${{ steps.prep.outputs.VERSION }}
+            org.opencontainers.image.created=${{ steps.prep.outputs.BUILD_DATE }}
+      - name: Check images
+        run: |
+          docker buildx imagetools inspect docker.io/${{ env.ORG }}/${{ env.CONTROLLER }}:${{ steps.prep.outputs.VERSION }}
+          docker pull docker.io/${{ env.ORG }}/${{ env.CONTROLLER }}:${{ steps.prep.outputs.VERSION }}
+      - name: Generate release manifests
```
Comment (aclevername) [inline .github/workflows/release.yaml:65]: what does this actually do? If the branch is tagged the files can't be overriden?

**Review by aclevername — APPROVED** [review]
lgtm for first pass, couple questions

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+          mkdir -p config/release
+          kustomize build ./config/crd > ./config/release/${{ env.CONTROLLER }}.crds.yaml
+          kustomize build ./config/manager > ./config/release/${{ env.CONTROLLER }}.deployment.yaml
+      - name: Do release
+        uses: softprops/action-gh-release@v1
+        with:
+          bodyPath: "docs/release_notes/${{ VERSION }}.md"
+          files: "config/release/*.yaml"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (aclevername) [inline .github/workflows/release.yaml:76]: what does this actually do? If the branch is tagged the files can't be overriden?

Comment (Callisto13) [reply .github/workflows/release.yaml:76]: what files?

Comment (aclevername) [reply .github/workflows/release.yaml:76]: ohhh its artifacts, nm I'm silly

Comment (Callisto13) [reply .github/workflows/release.yaml:76]: yeh but the job is creating a dir to hold the out files before upload to release. i don't think anything is being overwritten 🤷‍♀️ 

copied a bunch from flux

Comment (aclevername) [reply .github/workflows/release.yaml:76]: I guess my question would be should these files be committed to the tag branch, if flux don't then who cares :shrug:

**Pull request merged**